base/uk.ac.stfc.isis.ibex.opis/resources/hrpd_graph.py

```diff
@@ -0,0 +1,57 @@
+from org.csstudio.opibuilder.scriptUtil import PVUtil
+from java.lang import Thread, Runnable
+from org.csstudio.ui.util.thread import UIBundlingThread
+from org.csstudio.opibuilder.scriptUtil import ConsoleUtil
+from org.eclipse.swt.widgets import Display
+
+
+class GraphListener(Runnable):
+
+    def __init__(self):
+
+        self.graph_source_to_pv_map = {
+            "Sensor 1": {"VAL": 1, "EGU": 2, "PV": "$(P)CS:SB:DSC_CERNOX_1"},
```

Comment:
The PV field has now been removed, it was left over from when I didn't realise I could call getName. Though I agree we could do it programmatically I think that it makes the code less readable, see below for an example of what it would look like. I also think the use of a dictionary rather than tuple makes it more readable again.

```
sensors = {"Sensor {}".format(i): (i*2-1, i*2) for i in range(0+1, 4+1) }
differentials = {"Differential {}".format(i): (i*2+9, i*2+10) for i in range(0+1, 4+1) }
self.graph_source_to_pv_map = sensors + differentials
```

Reply:
Ok, in that case could you pull out "VAL" and "EGU" into constants at the top of the file?

Reply:
This no longer applies as I have moved to rules and tick boxes rather than the script and a drop down